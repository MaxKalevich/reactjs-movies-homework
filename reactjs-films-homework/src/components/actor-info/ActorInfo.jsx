import ActorPhoto from "../actor-photo/ActorPhoto";

import styles from "./actor-info-style.module.scss";

const ActorInfo = ({
  name,
  photo,
  birthday,
  placeOfBirthday,
  biography,
  actorImages,
}) => {
  return (
    <>
      <ActorPhoto actorPhoto={photo} />
      <div className={styles.contentContainer}>
        <h2 className={styles.name}>{name}</h2>
        <p>
          <span className={styles.birthday}>Birthday</span> {birthday}
        </p>
        <p className={styles.birthdayPlace}>
          <span className={styles.placeOfBirth}>Place Of Birth</span>
          {placeOfBirthday}
        </p>
        <p className={styles.biography}>
          <span className={styles.biographyHeader}>Biography</span>
          {biography
            ? biography
            : `Sorry, we don't have any information about ${name}`}
        </p>
        <div>
          <h2 className={styles.headerPhotos}>Photos</h2>
          <div className={styles.photosContainer}>
            <ActorPhoto actorPhotoCard={true} actorPhoto={actorImages} />
          </div>
        </div>
      </div>
    </>
  );
};

export default ActorInfo;
