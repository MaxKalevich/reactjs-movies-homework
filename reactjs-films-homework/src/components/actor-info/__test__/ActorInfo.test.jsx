import { render, screen } from "@testing-library/react";
import ActorInfo from "../ActorInfo";

describe("Checking ActorInfo component", () => {
  it("snapshot", () => {
    const { asFragment } = render(<ActorInfo />);
    expect(asFragment(<ActorInfo />)).toMatchSnapshot();
  });
  it("Contains text 'Birthday'", () => {
    render(<ActorInfo />);
    const linkElement = screen.getByText(/Birthday/i);
    expect(linkElement).toBeInTheDocument();
  });
  it("Contains text 'Place Of Birth'", () => {
    render(<ActorInfo />);
    const linkElement = screen.getByText(/Place Of Birth/i);
    expect(linkElement).toBeInTheDocument();
  });
  it("Contains text 'Biography'", () => {
    render(<ActorInfo />);
    const linkElement = screen.getByText(/Biography/i);
    expect(linkElement).toBeInTheDocument();
  });
  it("Contains text 'Photos'", () => {
    render(<ActorInfo />);
    const linkElement = screen.getByText(/Photos/i);
    expect(linkElement).toBeInTheDocument();
  });
});
