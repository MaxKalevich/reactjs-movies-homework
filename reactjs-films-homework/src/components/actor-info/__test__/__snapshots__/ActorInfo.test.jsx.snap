// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking ActorInfo component snapshot 1`] = `
<DocumentFragment>
  <img
    alt="actor"
    class="banner"
    src="movie-picture-mock.jpg"
  />
  <div
    class="contentContainer"
  >
    <h2
      class="name"
    />
    <p>
      <span
        class="birthday"
      >
        Birthday
      </span>
       
    </p>
    <p
      class="birthdayPlace"
    >
      <span
        class="placeOfBirth"
      >
        Place Of Birth
      </span>
    </p>
    <p
      class="biography"
    >
      <span
        class="biographyHeader"
      >
        Biography
      </span>
      Sorry, we don't have any information about undefined
    </p>
    <div>
      <h2
        class="headerPhotos"
      >
        Photos
      </h2>
      <div
        class="photosContainer"
      >
        <img
          alt="actor"
          class="banner"
          src="movie-picture-mock.jpg"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
